### ISSUE TITLE
Pre-work Checklist: Developer: Dustin Thomson

### ISSUE THREAD

**Issue by dustinowen** [hackforla/website#5724]
### Prerequisite
We are looking forward to having you on our team. Please make sure to attend the general Hack for LA onboarding to get the process started https://meetup.com/hackforla/events.

### Overview
As a new developer on the HfLA website team, fill in the following fields as you complete each onboarding item.

### Special Notes
1. It may take you a few weeks to finish this issue, because part of it is learning how to provide updates on issues that take more than a week to complete.    Please keep this issue open until you have been able to do all the steps.  
2. Usually we don't want to you have more than one issue assigned to you at a time, this issue is the exception, because it is instructions on how to work on other issues.
3. The action items listed below should mostly be worked on in a sequential order. However, you don't have to wait on one if you can proceed with the others, For instance, you don't have to wait for attending a weekly meeting before setting up your dev environment.  

### Action Items

- [x] Add yourself to the #hfla-site and #hfla-site-pr Slack channels
  - [x] To find contact information for the merge team members and technical leads, please take a look at our [Meet the Team wiki page](https://github.com/hackforla/website/wiki/Meet-the-Team)
- [x] Also, confirm with a merge team member or a technical lead that they have added you to the Google Calendar invites for our Zoom meetings
- [x] Add yourself to the [team roster](https://docs.google.com/spreadsheets/d/11u71eT-rZTKvVP8Yj_1rKxf2V45GCaFz4AXA7tS_asM/edit#gid=0)
- [x] After you have finished adding yourself to the roster, let a merge team member or a technical lead know you have added yourself to the roster and would like to be added to the `website-write` and `website` teams on GitHub
- [x] Self Assign this issue (gear in right side panel)
- [x] Add the "role: front end" or "role: back end" or both label(s) to this issue and remove the "role missing" label (gear in right side panel)
- [x] Add this issue to the Project Board under the Projects section (gear in right side panel)
- [x] Signup for a [Figma](https://Figma.com) account
- [x] Attend weekly team meetings:
  - [x] Developer (front-end/back-end) weekly team meeting, Tuesdays 7-8pm PST
  - [x] (Optional) Office Hours, Thursdays 7-8pm PST
  - [x] All team meeting (UX, Development, Product), Sunday 10am-12pm PST
- [x] Note: There are no meetings on the 1st-7th of every month.
- [x] Note regarding weekly team meeting requirements: All website team members are required to attend at least 1 team meeting in a week (held on Tuesdays, Thursdays and Sundays). In case, you are unable in any given week, you should reach out to the tech leadership team. Exceptions to this requirement may be provided on a case-by-case basis. Also, please let the tech leadership team know (through a slack message in the hfla-site channel as well as an @ mention in a comment of the issue that you would be working on) if you are planning to take a week off or a longer vacation.
- [x] Complete steps 1.1 - 1.6 in [Part 1: Setting up the development environment within Contributing.md](https://github.com/hackforla/website/blob/gh-pages/CONTRIBUTING.md#part-1-setting-up-the-development-environment)
  - [x] OPTIONAL: If you run into any issues, use [4.1 How do I ask for help within Contributing.md](https://github.com/hackforla/website/blob/gh-pages/CONTRIBUTING.md#41-what-do-i-do-if-i-need-help) as a visual example to get a hold of our [HfLA website leadership team](https://github.com/hackforla/website/projects/7#card-69730135). That way we can help you resolve any set up issues immediately.
- [x] Read section 2.1 - 2.4 in [Part 2: How the Website team works with GitHub issues within Contributing.md](https://github.com/hackforla/website/blob/gh-pages/CONTRIBUTING.md#part-2-how-the-website-team-works-with-github-issues)
  - [x] Follow the steps in section [2.3 Where can I find GitHub issues to work on?](https://github.com/hackforla/website/blob/gh-pages/CONTRIBUTING.md#23-where-can-i-find-github-issues-to-work-on) to assign yourself your first issue under the Project Board's Prioritized Backlog column and use [2.7 Working on a Issue within Contributing.md](https://github.com/hackforla/website/blob/gh-pages/CONTRIBUTING.md#27-working-on-an-issue) to start working on your issue
  - [x] Read [2.6 What to do when you need to stop mid issue](https://github.com/hackforla/website/blob/gh-pages/CONTRIBUTING.md#26-what-to-do-when-you-need-to-stop-mid-issue)
- [x] Once you take a good first issue, you will provide estimates and progress reports. Also, once you finish providing your "Estimate" action item below, please answer the question in the "Time spent so far" action item (also below).
   - [x] Estimate: Copy the below and put it in the "good first issue" that you picked.
 Check this box when you have completed this task so that we can identify if you understood the instructions and know what to do on all subsequent issues upon assignment. (Note: Please provide estimates on all the issues that you pick up, going forward.)
    ```
   Availability for this week:
   My estimated ETA for completing this issue:
    ```
   - [x] Time spent so far: Copy the question below into a new comment below (in this pre-work issue only) and answer it. This is just to get feedback on how long it took you to get to this point on this pre-work. There is no right or wrong answers. There is no judgement. It is ok if you take a long time or if you do it really fast or at any pace.
    ```
    How many hours did it take you to finish the pre-work up to and including adding your initial ETA and availability for your good first issue, including attending your first meetings?
    ```
   - [x] Progress Reports: Copy the below and put it in the issue once you have been assigned to the issue at least 5 days (we check weekly on Fridays), or sooner if you have something to report.  If you finish this issue before 5 days are reached, Yeah!!, do it on your next issue. **This update should be done every week for every issue that you are assigned to**. The checkbox here is meant for us to see if you understood the instructions when you end up doing your first weekly progress update.
    ```
    Provide Update
    1. Progress
    2. Blockers
    3. Availability
    4. ETA
    ```
- [x] Read sections 3.1.a - 3.1.c in [3.1 How to make a pull request](https://github.com/hackforla/website/blob/2e3c45fede99e61f479548ccca3f0dda83e351db/CONTRIBUTING.md#31-how-to-make-a-pull-request) to learn more about how to make a pull request (PR) for the issue that you are working on and how to make changes to your PR if changes are requested by the reviewer
- [x] Additional notes:
   - [x] Please don't forget to review pull requests whenever you are able to.  The [How to review Pull Requests](https://github.com/hackforla/website/wiki/How-to-review-pull-requests) guide will teach you how to review pull requests.
   - [x] Please work on only one issue at a time and wait until your pull request is merged before picking up another issue.
- [x] Read and understand how we progress through issues. Then, you can check this off.
Progress through issues in the [prioritized backlog](https://github.com/hackforla/website/projects/7#column-7198257) only with increasing complexity in the following order:
  - Good first issue (two per person)
  - Small (one per person, with some exceptions, see below)
  - Medium (you can work on more than one medium issue, but only one at a time)
  - Large (you can work on more than one large issue, but only one at a time)
    - The reasons for this progression are:
      - The issues start out as being prescriptive and become less so as you gain more experience by working through increasingly complex issues.
      - We are trying to teach you the team methodology through the issues themselves.
      - It ensures you understand what we expect and the quality of contributions.
    - You can work on back-to-back small issues if it meets the following criteria:
      - You are learning something new and need to work on an issue of a lesser complexity
      - Special request by a lead or pm
- [x] Read the [Start Here - Developers](https://www.figma.com/file/0RRPy1Ph7HafI3qOITg0Mr/Hack-for-LA-Website?node-id=8583%3A0) in Figma
- [x] Go familiarize yourself with the [Hack for LA Design System page in Figma](https://www.figma.com/file/0RRPy1Ph7HafI3qOITg0Mr/Hack-for-LA-Website?node-id=3464%3A3)  (where you can see components and their SCSS classes)
- [x] Once all tasks are completed, close this issue.  This issue will remain open until you get through your first progress report.  Once you have done that, we are confident you know how to keep the momentum going on your issue and keep your team informed.

### What should I do if I have a question about an issue I'm working on, and I haven't gotten a response yet?
- First, you should post the question or blocker as a comment on your assigned issue, so it can be easily referred to in the next bullet points.
- Then, add the label "Status: Help Wanted" so other developers can see it and potentially help answer your question. In addition, you will still need to post a Slack message or bring it up in meeting so we know you need help; see below for how to do that.
- Also, you can post your question on the hfla-site slack channel and link the issue you're working on, so other developers can see and respond.
- Lastly, you can add the issue to the "Development team meeting discussion items" column of the Project Board so that it can be addressed in the next development meeting. Please bring it during the meeting that you need help.

### Resources/Instructions
- [Contributing.md - Hack for LA](https://github.com/hackforla/website/blob/gh-pages/CONTRIBUTING.md)
- [GitHub Project Board - Hack for LA](https://github.com/hackforla/website/projects/7)
- [GitHub Project Board Prioritized Backlog highlighted and sorted by Front End](https://github.com/hackforla/website/projects/7?card_filter_query=label%3A%22role%3A+front+end%22)
- [GitHub Project Board Prioritized Backlog highlighted and sorted by Back End/DevOps](https://github.com/hackforla/website/projects/7?card_filter_query=label%3A%22role%3A+back+end%2Fdevops%22)
- [Figma - Hack for LA](https://www.figma.com/file/0RRPy1Ph7HafI3qOITg0Mr/Hack-for-LA-Website)
- [Google Drive - Hack for LA website team](https://drive.google.com/drive/folders/1p76K0FgfiAWeIIEyoyJ_Iik8FVj8cBjT?usp=sharing)
- [Agenda / Notes - Dev Team Tuesday meeting](https://github.com/hackforla/website/issues/2010)
- [Agenda / Notes - All Team meeting](https://github.com/hackforla/website/issues/2027)
- [How to review Pull Requests](https://github.com/hackforla/website/wiki/How-to-review-pull-requests)

**Comment by dustinowen** [comment]
**How many hours did it take you to finish the pre-work up to and including adding your initial ETA and availability for your good first issue, including attending your first meetings?**

Hello! After onboarding I paused for a bit as I had a trip planned to Yosemite and ended up staying longer than planned. I am now able to dedicate the required hours (plus more!) to HackForLA. 

I dove back in to the CONTRIBUTING.md today to get everything set up and complete my 'good first issue'. I have spent a little over an hour completing up to this point. 

I do want to apologize as I saw a note about communicating vacations (which I did not do). I have made note of this and will be more communicative with the team going forward. I will be honest I was a little apprehensive with jumping in just as I am a "fresher" and needed to digest this all. I am happy to report I am excited to be contributing and fully committed to being a part of this team. 

- I think I may have submitted my pull request wrong. I followed a pattern I have used in the past but am unsure if it met the expectations for what HackforLA needs. I will await feedback but do want to make sure I understand this process completely.

**Comment by JessicaLucindaCheng** [comment]
@dustinowen Please self-assign yourself to this pre-work.

**Comment by dustinowen** [comment]
@JessicaLucindaCheng @roslynwythe 

Question regarding 'Progress Reports' - 

_If you finish this issue before 5 days are reached, Yeah!!, do it on your next issue. This update should be done every week for every issue that you are assigned to._ 

This says if you finish before 5 days, do it on your next issue. But also says it should be done for every issue from the week you are assigned to. I am wanting to be sure I understand the guidelines you are expecting here...

Is this is to be done on all issues from the week, even if the issue is completed/closed?

**Comment by JessicaLucindaCheng** [comment]
> @JessicaLucindaCheng @roslynwythe
> 
> Question regarding 'Progress Reports' -
> 
> _If you finish this issue before 5 days are reached, Yeah!!, do it on your next issue. This update should be done every week for every issue that you are assigned to._
> 
> This says if you finish before 5 days, do it on your next issue. But also says it should be done for every issue from the week you are assigned to. I am wanting to be sure I understand the guidelines you are expecting here...
> 
> Is this is to be done on all issues from the week, even if the issue is completed/closed?

No, you don't have to do it on completed/closed issues. 

Here is the overall process:
1) For every issue, you provide an initial availability and ETA upon self-assignment. 
2) Then, if your issue is still open and a week has passed since you provided an initial availability and ETA, you will provide a progress report that includes progress, blockers, availability, and ETA.
3) If another week has passed since your last progress report and your issue is still open, you will provide another progress report.
4) Repeat step 3 until your issue is closed.

**Comment by HackforLABot** [comment]
## Activity Log: dustinowen
### Repo: https://github.com/hackforla/website

#####  ⚠ Important note: The bot updates this comment automatically - do not edit

- dustinowen opened: https://github.com/hackforla/website/issues/5724 at 10/16/2023, 07:48 PM PDT
- dustinowen assigned: https://github.com/hackforla/website/issues/5790 at 11/01/2023, 01:58 PM PDT
- dustinowen opened: https://github.com/hackforla/website/pull/5820 at 11/01/2023, 02:17 PM PDT
- dustinowen commented on: https://github.com/hackforla/website/issues/5790#issuecomment-1789704480 at 11/01/2023, 02:19 PM PDT
- dustinowen commented on: https://github.com/hackforla/website/issues/5724#issuecomment-1789715455 at 11/01/2023, 02:29 PM PDT
- dustinowen closed: https://github.com/hackforla/website/pull/5820#event-10867329522 at 11/05/2023, 07:50 PM PST
- dustinowen opened: https://github.com/hackforla/website/pull/5858 at 11/05/2023, 08:07 PM PST
- dustinowen commented on: https://github.com/hackforla/website/pull/5820#issuecomment-1794053157 at 11/05/2023, 08:07 PM PST
- dustinowen commented on: https://github.com/hackforla/website/pull/5858#issuecomment-1794054507 at 11/05/2023, 08:09 PM PST
- dustinowen commented on: https://github.com/hackforla/website/pull/5858#issuecomment-1801008214 at 11/07/2023, 08:08 PM PST
- dustinowen PR was merged: https://github.com/hackforla/website/pull/5858#event-10895747242 at 11/07/2023, 09:25 PM PST
- dustinowen issue closed by PR 5858: https://github.com/hackforla/website/issues/5790#event-10895747363 at 11/07/2023, 09:25 PM PST
- dustinowen closed by PR 5917: https://github.com/hackforla/website/issues/5876#event-10960153397 at 11/14/2023, 01:11 PM PST
- dustinowen assigned: https://github.com/hackforla/website/issues/5874 at 11/14/2023, 01:19 PM PST
- dustinowen commented on: https://github.com/hackforla/website/issues/5874#issuecomment-1811334326 at 11/14/2023, 01:21 PM PST
- dustinowen opened: https://github.com/hackforla/website/pull/5903 at 11/14/2023, 01:40 PM PST
- dustinowen PR was merged: https://github.com/hackforla/website/pull/5903#event-10973045721 at 11/15/2023, 02:33 PM PST
- dustinowen issue closed by PR 5903: https://github.com/hackforla/website/issues/5874#event-10973046048 at 11/15/2023, 02:33 PM PST
- dustinowen assigned: https://github.com/hackforla/website/issues/5855 at 11/15/2023, 07:22 PM PST
- dustinowen commented on: https://github.com/hackforla/website/issues/5855#issuecomment-1813731451 at 11/15/2023, 07:24 PM PST
- dustinowen opened: https://github.com/hackforla/website/pull/5911 at 11/15/2023, 08:17 PM PST
- dustinowen assigned: https://github.com/hackforla/website/issues/5724 at 11/16/2023, 01:31 PM PST
- dustinowen commented on: https://github.com/hackforla/website/pull/5911#issuecomment-1815467922 at 11/16/2023, 03:15 PM PST
- dustinowen commented on: https://github.com/hackforla/website/issues/5724#issuecomment-1815491798 at 11/16/2023, 03:28 PM PST
- dustinowen PR was merged: https://github.com/hackforla/website/pull/5911#event-10999505098 at 11/17/2023, 01:56 PM PST
- dustinowen issue closed by PR 5911: https://github.com/hackforla/website/issues/5855#event-10999505285 at 11/17/2023, 01:56 PM PST
- dustinowen closed as completed: https://github.com/hackforla/website/issues/5724#event-11090004913 at 11/28/2023, 03:57 PM PST
